src/border_agent/border_agent.cpp

```diff
@@ -157,12 +163,13 @@ BorderAgent::BorderAgent(otbr::Ncp::RcpHost &aHost, Mdns::Publisher &aPublisher)
     : mHost(aHost)
     , mPublisher(aPublisher)
     , mIsEnabled(false)
-    , mIsEphemeralKeyEnabled(false)
+    , mIsEphemeralKeyEnabled(OTBR_CONFIG_BORDER_AGENT_EPHEMERAL_KEY_ON_INIT)
```

Comment:
do we need the macro `OTBR_CONFIG_BORDER_AGENT_EPHEMERAL_KEY_ON_INIT`?
```suggestion
    , mIsEphemeralKeyEnabled(otThreadGetVersion() >= OT_THREAD_VERSION_1_4)
```

Reply:
Use this simple way for this PR; Link [detailed discussion](https://docs.google.com/document/d/1mpNWtLGcO_n-EmvE9YryFvevcCI0fdYZfWM8faV5HGI/edit?usp=sharing&resourcekey=0-8ZRqr7Fm8qnHGUITw3SeQQ) for ref